### PR TITLE
Adds a generic Terraform task which uses mapped variables

### PR DESCRIPTION
**Pull request by ElvenSpellmaker**
* Adds a generic Terraform task which performs init/plan/apply and outputs variables
 * Refactors the old AKS template to use the new generic template

Note this needs testing.

### CODE REVIEW

**Comment by guibirow** [comment]
Does it fix any issue? Could link the issue in the PR so we can identify what is the reason for this change.
Thanks

**Comment by ElvenSpellmaker** [comment]
@guibirow I'm not sure if there's an issue, it's something @williamayerst asked to be done.

**Comment by williamayerst** [comment]
@guibirow  the reason for this change is so that we can have a single unified terraform template for deploying any kind of resource, rather than needing to create unique steps/jobs for each grouping of resources (i.e. for AKS we need to supply cluster size, so we would need to support it directly in the step for every single TF deployment job).

This way, we reduce duplication.

**Review by williamayerst — APPROVED** [review]

File: azDevOps/azure/templates/jobs/deploy-aks.yml [hunk 1]
```diff
@@ -9,10 +9,10 @@ parameters:
   terraform_state_container: ''
   terraform_state_key: ''
   terraform_state_workspace: ''
-  company: ''
-  project: ''
+  company: 'amido'
+  project: 'stacks'
   # Azure Config
-  location: ''
+  location: 'uksouth'
```
Comment (williamayerst) [inline azDevOps/azure/templates/jobs/deploy-aks.yml:15]: These shouldn't be default values IMO

**Pull request merged**